resourcet/Data/Acquire/Internal.hs

```diff
@@ -25,8 +27,34 @@ import qualified Control.Monad.Catch as C ()
 -- @since 1.1.2
 data ReleaseType = ReleaseEarly
                  | ReleaseNormal
-                 | ReleaseException
-    deriving (Show, Read, Eq, Ord, Enum, Bounded, Typeable)
+                 | ReleaseException' E.SomeException
+    deriving (Show, Typeable)
+
+-- | Treats 'E.SomeException's as equal when they wrap the same type and 'show' the same.
+instance Eq ReleaseType where
```

Comment:
I think I'm more tempted to simply bypass the instance entirely rather than rely on a potentially finicky `Show` instance, thoughts?

Reply:
Done